### PR TITLE
Rewrite stake tab in vue

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract

Rewrite stake tab in vue

## Testing
Test that the stake functionality works

## TODO
- [x] Write unit tests
- [x] Remove old stake global variables

### CODE REVIEW

**Comment by BreadJS** [comment]
Wen merge?^-^

**Comment by Duddino** [comment]
Those errors were mostly due to the two `useWallet` instances not being in sync. I've fixed it using the pinia library, which uses a very similar structure to what we were using previously, the only change code-wise is that we have to drop the `.value` when using properties from `useWallet`

**Comment by JSKitty** [comment]
Upon un-delegating (straight after delegation, via internal mempool) I'm getting this `getAddress is not a function` error, presumably a change address generation issue.

```
Uncaught (in promise) TypeError: getAddress is not a function
    unstake Stake.vue:72
    callWithErrorHandling runtime-core.esm-bundler.js:195
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:202
    emit runtime-core.esm-bundler.js:715
    node_modules mpw.js:10759
    submit StakeInput.vue:35
    1].rules[1]!./node_modules/vue-loader/dist/index.js??ruleSet[0].use!./scripts/stake/StakeInput.vue?vue&type=template&id=1bed7ffa/render/<.default</_cache[4 StakeInput.vue:158
    callWithErrorHandling runtime-core.esm-bundler.js:195
    callWithAsyncErrorHandling runtime-core.esm-bundler.js:202
    invoker runtime-dom.esm-bundler.js:693
```

image

**Comment by JSKitty** [comment]
Staking/Unstaking is working well now it seems, balances accurate, activity displaying well, prompts look good.
Last two minor regressions I see regarding the "Set Cold Address" dialog is:
- Auto-focus was lost.
- The user no longer can give an "empty input", which used to make MPW use the Cold Pool defined in chainparams; it's slightly more difficult to return to the default address now.

After this I think we're ready for Quality Control.

**Comment by JSKitty** [comment]
Found a niche bug, with "Advanced Mode" enabled, trying to set a custom Owner Address from a Contact (mouse on button) results in this:

```
Uncaught (in promise) TypeError: domInput is undefined
    guiSelectContact contacts-book.js:286
    onclick 
```

image

**Review by JSKitty — DISMISSED** [review]
**tACK** 9f0c15c2413177518358c6daf52698520a527ebc

Pushing to Quality Control, confident in the PR's stability at this point.

**Comment by JSKitty** [comment]
> Found a niche bug, with "Advanced Mode" enabled, trying to set a custom Owner Address from a Contact (mouse on button) results in this:
> 
> ```
> Uncaught (in promise) TypeError: domInput is undefined
>     guiSelectContact contacts-book.js:286
>     onclick 
> ```
> 
> image

Relevant to this niche issue: I noticed the Stake dialog (advanced Owner Address) after clicking or typing a Contact's name, cannot send the transaction as the Name is not being converted to the Address it's associated to, for this, you can use the same name->address lookup that the normal Send flow uses.

**Review by JSKitty — CHANGES_REQUESTED** [review]

File: scripts/wallet.js [hunk 1]
```diff
@@ -811,7 +810,7 @@ export class Wallet {
     subscribeToNetworkEvents() {
         getEventEmitter().on('new-block', async (block) => {
             if (this.#isSynced) {
-                stakingDashboard.update(0);
+                await getNetwork().getLatestTxs(this);
```
Comment (panleone) [inline scripts/wallet.js:813]: getLatestTxs must not be called anymore after first sync
`await this.getLatestBlocks(block);` will handle both

**Review by panleone — CHANGES_REQUESTED** [review]

File: scripts/composables/use_wallet.js [hunk 2]
```diff
@@ -49,6 +50,7 @@ export function useWallet() {
         await wallet.checkDecryptPassword(passwd);
 
     hasEncryptedWallet().then((r) => {
+        console.log(r);
```
Comment (JSKitty) [inline scripts/composables/use_wallet.js:53]: Nuke

File: scripts/parsed_secret.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                test = rule.test(secret, password);
+            } catch (e) {
+                test = false;
+            }
+            if (test) {
+                return await rule.f(secret, password);
+            }
+        }
+        return null;
+        // createAlert('warning', ALERTS.FAILED_TO_IMPORT + '<br>', 6000);
```
Comment (JSKitty) [inline scripts/parsed_secret.js:124]: Nuke? I assume callers of `parse` will handle any `null` return notifications.

File: scripts/stake/Stake.vue [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        return;
+    }
+    const cDB = await Database.getInstance();
+    const cAccount = await cDB.getAccount();
+    const returnAddress =
+        cAccount?.getContactBy({
+            name: ownerAddress,
+            pubkey: ownerAddress,
+        })?.pubkey || ownerAddress;
+    console.log(returnAddress, ownerAddress);
```
Comment (JSKitty) [inline scripts/stake/Stake.vue:61]: Nuke

File: scripts/stake/StakeInput.vue [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+function syncAmount() {
+    if (amountCurrency.value === '') {
+        amount.value = '';
+    } else {
+        amount.value = amountCurrency.value / price.value;
+    }
+}
+
+async function selectContact() {
+    console.log(ownerAddress.value);
```
Comment (JSKitty) [inline scripts/stake/StakeInput.vue:62]: Nuke

File: scripts/stake/StakeInput.vue [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    ownerAddress.value = (await promptForContact()) || '';
+    console.log(ownerAddress.value);
```
Comment (JSKitty) [inline scripts/stake/StakeInput.vue:64]: Nuke

**Review by JSKitty — DISMISSED** [review]
**tACK** bbe36d1c2c1a1706e5dfe2ea621fb1b679eb79f9

LGTM, just a few logs to nuke then we should merge this ASAP.

**Review by JSKitty — APPROVED** [review]
**re-tACK**

**Review by panleone — APPROVED** [review]
tACK d77c2372ac3268bfe787c2a93ba1c63841bfca33

**Pull request merged**